### DISCUSSION TITLE
How to prevent any button triggering validation?

### DISCUSSION THREAD

**Discussion by vinamelody** [react-hook-form/react-hook-form#2557] [Q&A] [answered]
Hi, if I have more than 1 button in my form component, somehow the validation triggers on any button click
i've tried to only define 1 button type `submit`.
For example, here's the codesandbox link: https://codesandbox.io/s/stupefied-borg-zvl16?file=/src/App.js

If you click on Back button or Choose File button, the error message `Required` is immediately shown for other fields.

Thanks for the help! I'm still learning react-hook-form 🙏🏼 ..

**Comment by bluebill1049** [comment, marked as answer]
i think you miss type="button"

**Reply by vinamelody** [reply]
oh geez... it's that simple 😲, thanks a lot @bluebill1049

**Reply by bluebill1049** [reply]
all good 👍
